Adding MonoT5 finetuning script on MS MARCO GPU (#224)
castorini/pygaggle: docs/experiments-finetune-monot5-gpu.md

To do this, first clone the repository recursively.

```
git clone --recursive https://github.com/castorini/pygaggle.git
```

Then install PyGaggle using:

```
pip install --upgrade pygaggle/
```

## Model

+ monoT5-base: Document Ranking with a Pretrained Sequence-to-Sequence Model [(Nogueira et al., 2020)](https://arxiv.org/pdf/2003.06713.pdf)

## MonoT5 finetuning
### Data Prep

We're first going to download and extract the triples.small from MS MARCO.
```
wget https://msmarco.blob.core.windows.net/msmarcoranking/triples.train.small.tar.gz -P triples
tar -zxf triples/triples.train.small.tar.gz -C triples/
```
Then, create a new directory so the training script can save model checkpoints to.
```
mkdir monoT5_model/
```
### Train
To train the MonoT5 model, run the following command:
```
nohup python -um pygaggle.run.finetune_monot5 --triples_path triples/triples.train.small.tsv \
                                              --save_every_n_steps 10000 \
                                              --output_model_path monoT5_model/ &

> tail -f nohup

Reviewer: It should be nohup**.out**?

```
Here, we consider one epoch to be 6.4e5 of lines of the triples.train.small.tsv, which corresponds to 12.8e5 training examples (positives + negatives) and 10k steps using batch of 128. The number of epochs defaults to 10 (100k steps). Adding the "--epoch" argument with an integer will include `n * 6.4e5` lines to the training set.
If you don't wish to save checkpoints every N step, simply omit the --save_every_n_steps argument.

> With a Nvidia V100 it takes around 2~3 days to train the model on 10 epoch.

Reviewer: ```suggestion
With an Nvidia V100, it takes around 2~3 days to train the model for 10 epochs.
```
